### PR TITLE
HPCC-26003 Report a warning to NOTES.txt if ephemeral storage is used

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/hpcc/templates/NOTES.txt [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+{{/* Gather a list of ephemeral and persistant planes */}}
+{{- $storage := (.Values.storage | default dict) -}}
+{{- $match := dict "ephemeral" (list) "persistant" (list) -}}
+{{- $planes := ($storage.planes | default list) -}}
+{{- $searchLabels := list "data" "dali" "sasha" "dll" "lz" -}}
+{{- range $plane := $planes -}}
+{{- $labels := $plane.labels | default (list "data") -}}
+{{- $tag := (hasKey $plane "storageClass" | ternary "ephemeral" "persistant") }}
+{{- range $label := $labels -}}
+{{- if has $label $searchLabels -}}
```
Comment (jakesmith) [inline helm/hpcc/templates/NOTES.txt:13]: style: we're not consistent at all, but I think it makes it more readable if we indent scopes, e.g. from issuers.yaml :
```
spec:
  {{- if not .root.Values.global.noResourceValidation -}}
  {{- if .me.spec.ca }}
   {{- if .me.spec.ca.secretName }}
    {{- if not (lookup "v1" "Secret" .root.Release.Namespace .me.spec.ca.secretName) }}
      {{- $_ := fail (printf "\n\nUsing a local certificate authority requires a CA certificate stored in the secret named '%s'. To disable mTLS security set \"certificates.enabled=false\". To bypass this validation check set \"global.noResourceValidation=true\". \n" .me.spec.ca.secretName ) -}}
    {{- end }}
   {{- end }}
  {{- end }}
  {{- end }}
```

**Review by jakesmith — APPROVED** [review]
@ghalliday - logic looks good to me, I might indent template scopes for readibility

**Comment by ghalliday** [comment]
@jakesmith re-indented the file.  I think I prefer 1 space to 2 from trying both out.

**Review by jakesmith — APPROVED** [review]
@ghalliday - looks good

File: helm/hpcc/templates/NOTES.txt [hunk 2]
```diff
@@ -1,43 +1,65 @@
...
-{{-  range $serviceName, $_service := $sashaServices }}
-{{-   $service := ($_service | default dict) -}}
-{{-   if not $service.disabled -}}
-{{-    if hasKey $names $serviceName -}}
-{{-     required ( printf "Component name %s in section sasha was already seen in section %s" $serviceName (get $names $serviceName)) nil -}}
-{{-    else }}
+ {{- range $serviceName, $_service := $sashaServices }}
+  {{- $service := ($_service | default dict) -}}
+  {{- if not $service.disabled -}}
+   {{-  if hasKey $names $serviceName -}}
```
Comment (richardkchapman) [inline helm/hpcc/templates/NOTES.txt:54]: trivial: extra leading spaces after {{-

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**